butterfly-slack/src/main/java/com/paypal/butterfly/slack/SlackNotifierConfig.java

```diff
@@ -0,0 +1,133 @@
+            private WebTarget target;
+            private String postSuccessJson;
+            private String postAbortJson;
+            private boolean active = false;
+
+            @PostConstruct
+            public void postConstruct() {
+                try {
+                    token = System.getProperty("slack.token");
+                    if (token == null) {
+                        logger.error("Unable to locate slack token to send notifications");
+                        return;
+                    }
+                    Path postSuccessPath = new File(this.getClass().getResource("/postSuccess.json").toURI()).toPath();
+                    Path postAbortPath = new File(this.getClass().getResource("/postAbort.json").toURI()).toPath();
+                    postSuccessJson = new String(Files.readAllBytes(postSuccessPath));
+                    postAbortJson = new String(Files.readAllBytes(postAbortPath));
+                    Client client = ClientBuilder.newClient();
+                    target = client.target("https://www.slack.com/api/");
+                    active = true;
+                } catch (IOException e) {
+                    logger.error("Unable to read the slack message template file from the path", e);
+                } catch (URISyntaxException e) {
+                    logger.error("Unable to read path of the slack message template file", e);
+                }
+            }
+
+            @Override
+            public void postTransformation(TransformationRequest transformationRequest, TransformationResult transformationResult) {
+                if (active && postSuccessJson != null) {
+                    //TODO get slackUserEmails, details link and github url from butterfly wrapper
+                    String[] slackUserEmails = new String[]{"edwardhwang@berkeley.edu", "ehwang0824@gmail.com"};
+                    postNotification(postSuccessJson, slackUserEmails);
+                }
+            }
+
+            @Override
+            public void postTransformationAbort(TransformationRequest transformationRequest, TransformationResult transformationResult) {
+                if (active && postAbortJson != null) {
+                    //TODO get slackUserEmails from butterfly wrapper
+                    String[] slackUserEmails = new String[]{"edwardhwang@berkeley.edu", "ehwang0824@gmail.com"};
+                    postNotification(postAbortJson, slackUserEmails);
+                }
+            }
+
+            private void postNotification(String jsonMessage, String[] slackUserEmails) {
+                for (String email : slackUserEmails) {
+                    try {
+                        String slackUserId = getSlackUserId(token, email);
+                        postSlackMessage(token, jsonMessage, slackUserId);
+                    } catch (ProcessingException e) {
+                        logger.warn("An exception happened when retrieving the slack user id, web target url is invalid", e);
```

Comment:
There are many potential reasons to get a `ProcessingException` here. The URL being invalid is just one of them. Also, the cause could be when getting slack user id or when sending slack message. Please make the warning message more generic.